### ISSUE TITLE
jcmd help doesn't indicate how to use -l, -h, or -J

### ISSUE THREAD

**Issue by pdbain-ibm** [eclipse/openj9#6796]
The reference implementation shows:
```
Usage: jcmd <pid | main class> <command ...|PerfCounter.print|-f file>
   or: jcmd -l                                                    
   or: jcmd -h                                                    
```

Suggested changes to OpenJ9 `jcmd`:
```
Usage : jcmd <vmid> <target arguments>
    or:
       jcmd <jcmd arguments>

jcmd arguments: 
    -J : supply arguments to the Java VM running jcmd
    -l :  list JVM processes on the local machine
    -h : print this help message

<vmid> : Attach API VM ID as shown in jps or other Attach API-based tools.

target arguments:
    help : print the list of diagnostic commands
    help command : print help for the specific command
    command [command arguments] : command from the list returned by "help"
```

**Comment by pshipton** [comment]
@JasonFengJ9

**Comment by JasonFengJ9** [comment]
@pdbain-ibm `jcmd` argument `-J` works with `<vmid> <target arguments>`, how about
```
Usage : jcmd [options] [<vmid> <arguments>]
```

This format appears closer to `java command line option` [1].

[1] https://docs.oracle.com/en/java/javase/12/tools/java.html#GUID-3B1CE181-CD30-4178-9602-230B800D4FAE

**Comment by doveye** [comment]
I think the options should be in angle brackets, otherwise the syntax is suggesting you use the actual text "options". Hence I went with `jcmd [<options>] [<vmid> <arguments>]` for the first draft of the docs. It's not completely accurate, but I think it's clear enough once you understand the options (ie, you wouldn't try running `jcmd -h 12345 <command>`)

I actually played around a bit with this yesterday, but the full syntax got very complex! I considered the following as a possibility:

```
jcmd [-l|-h]
or
jcmd [-J<java_arg>, -J<java_arg>, ...] <vmid> <arguments>
```

Or perhaps the following. It's not totally clear what's on the right-hand side of the "or", but it should be fairly obvious once you understand the options:

```
jcmd [-l|-h] | [<java_arguments>] <vmid> <target_arguments>
```

**Comment by JasonFengJ9** [comment]
@doveye is there a guideline or convention about command line syntax, particularly for OpenJ9?

Following doc seem not consistent:

[1] https://eclipse.github.io/openj9-docs-staging/341/tool_jcmd/ 
```
jcmd [<options>] [<vmid> <arguments>]
```

[2] https://eclipse.github.io/openj9-docs-staging/341/tool_jstack/
```
jstack <options>* <pid>*
```

[3] https://eclipse.github.io/openj9-docs-staging/341/tool_jdmpview/
```
jdmpview (-core <core file> | -zip <zip file>) [-notemp]
```

**Comment by doveye** [comment]
@JasonFengJ9  There's no OpenJ9-specific convention, we just try to use what appears to be the universal standard (probably originated with POSIX - https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html, and probably what the IBM standards are based on), though I think we've departed from it a bit with these j-commands!

I can't see what's inconsistent about the jdmpview syntax, but the jstack line does look like it needs changing. From the description, I think it should be `jstack [<options>] [<pid>]`  

Mind you, the POSIX standard does say 

> Utilities with very complex arguments may be shown as follows:
> 
> utility_name [options][_operands_]

though unfortunately they don't elaborate on that. I see that just "options" is used in the syntax of the Hotspot tools, but then they also use italics (not mentioned in POSIX standard) and don't use angle brackets (are mentioned in the standard). Argh!

**Comment by JasonFengJ9** [comment]
> I can't see what's inconsistent about the jdmpview syntax

I was referring the `(` & `)` instead of `[` & `]`.
```
jdmpview (-core <core file> | -zip <zip file>) [-notemp]
```

**Comment by doveye** [comment]
Ok. The `()` are just there to clarify what the pipe applies to. The `[]` indicate an optional parameter.

**Comment by doveye** [comment]
No, I don't see `()` in the POSIX doc either. That doc says (in point 8) that you can use multiple synopsis lines instead. The use of `()` in jdmpview is just the basic one (eg in maths or text) of using brackets to separate items from what's outside the brackets.
